api/tests/Feature/Notifications/ApplicationDeadlineExtendedTest.php

```diff
@@ -0,0 +1,169 @@
+        parent::setUp();
+
+        $this->seed(RolePermissionSeeder::class);
+
+        $this->fixtureNotification = new ApplicationDeadlineExtended(
+            'userName',
+            Carbon::parse('2999-12-31'),
+            'poolNameEn',
+            'poolNameFr',
+            '1',
+        );
+    }
+
+    // respects each user's notification preferences
+    public function testRespectsNotificationPreferencesNone(): void
+    {
+        $user = User::factory()
+            ->create([
+                'enabled_email_notifications' => [NotificationFamily::APPLICATION_UPDATE->name],
+                'enabled_in_app_notifications' => [NotificationFamily::APPLICATION_UPDATE->name],
+            ]);
+        assertEqualsCanonicalizing([GcNotifyEmailChannel::class, 'database'], $this->fixtureNotification->via($user));
+    }
+
+    // respects each user's notification preferences
+    public function testRespectsNotificationPreferencesBoth(): void
+    {
+        $user = User::factory()
+            ->create([
+                'enabled_email_notifications' => [NotificationFamily::JOB_ALERT->name],
+                'enabled_in_app_notifications' => [NotificationFamily::JOB_ALERT->name],
+            ]);
+        assertEquals([], $this->fixtureNotification->via($user));
+    }
+
+    // respects each user's notification preferences
+    public function testRespectsNotificationPreferencesJustEmail(): void
+    {
+        $user = User::factory()
+            ->create([
+                'enabled_email_notifications' => [NotificationFamily::APPLICATION_UPDATE->name],
+                'enabled_in_app_notifications' => [NotificationFamily::JOB_ALERT->name],
+            ]);
+        assertEquals([GcNotifyEmailChannel::class], $this->fixtureNotification->via($user));
+    }
+
+    // respects each user's notification preferences
+    public function testRespectsNotificationPreferencesJustInApp(): void
+    {
+        $user = User::factory()
+            ->create([
+                'enabled_email_notifications' => [NotificationFamily::JOB_ALERT->name],
+                'enabled_in_app_notifications' => [NotificationFamily::APPLICATION_UPDATE->name],
+            ]);
+        assertEquals(['database'], $this->fixtureNotification->via($user));
+    }
+
+    // Build the notification and send it to the live GC Notify service
+    public function testCanSendGcNotify(): void
+    {
+        if (! config('notify.client.apiKey')) {
+            $this->markTestSkipped('API key not found');
+        }
+
+        $user = User::factory()
+            ->create([
+                'email' => config('notify.smokeTest.emailAddress'),
+                'enabled_email_notifications' => [NotificationFamily::APPLICATION_UPDATE->name],
+                'enabled_in_app_notifications' => [NotificationFamily::JOB_ALERT->name],
+            ]);
+
+        $user->notify($this->fixtureNotification);
+        $exceptionThrown = false;
+
+        assertFalse($exceptionThrown);
```

Comment:
This is a bit of a weak assertion.  You can use the facade to directly confirm if a notification was sent or not.
https://laravel.com/docs/11.x/notifications#testing

Reply:
Very nice.